### DISCUSSION TITLE
Linux about the badblocks command source code, I want to learn

### DISCUSSION THREAD

**Discussion by IDLON** [util-linux/util-linux#2071] [Q&A] [answered]
I was able to find the command, but I couldn't find the source code for it

**Comment by t-8ch** [comment, marked as answer]
It's part of e2fsprogs, not util-linux:
https://github.com/tytso/e2fsprogs/blob/67f2b54667e65cf5a478fcea8b85722be9ee6e8d/misc/badblocks.c

**Reply by IDLON** [reply]
Many thanks！！！
